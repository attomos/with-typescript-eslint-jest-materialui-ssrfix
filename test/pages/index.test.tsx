import React from "react";
import { render, fireEvent } from "../testUtils";
import IndexPage from "../../pages/index";

describe("IndexPage", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<IndexPage />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  it("clicking button triggers alert", () => {
    const { getByText } = render(<IndexPage />, {});
    window.alert = jest.fn();
    fireEvent.click(getByText("test"));
    expect(window.alert).toHaveBeenCalledWith("With typescript and Jest");
  });
});
