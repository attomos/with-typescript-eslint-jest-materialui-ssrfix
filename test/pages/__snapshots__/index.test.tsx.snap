// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <header>
      <nav>
        <a
          href="/"
        >
          Home
        </a>
         | 
        <a
          href="/users"
        >
          Users List
        </a>
         | 
        <a
          href="/api/users"
        >
          Users API
        </a>
      </nav>
    </header>
    <h2>
      Helllo
    </h2>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        test
      </span>
    </button>
    <footer>
      <hr />
      <span>
        I'm here to stay (Footer)
      </span>
    </footer>
  </div>
</DocumentFragment>
`;
